### PR TITLE
Fix #252, Integrate SMS in workflow, generate enclaveChallenge for TEE task.

### LINKED ISSUE #252: Update worker infos when changing the machine and keeping the same wallet
Expected behaviour: when we change the worker machine and keep the same wallet, the worker description (cpu, ram, tee) should be updated.

Actual behaviour: Those informations are not changed (we keep the existing ones)

### PR DESCRIPTION
**Pull request by Zied-Guesmi**
- Add spring feign-client to communicate with resultRepo and SMS
- Update worker configuration when reconnecting (fixes #252 )
- Send SMS IP to worker
- Get generated EnclaveChallenge from SMS and send it to worker.
- Use spring-retry to handle failure of http requests.

### CODE REVIEW

File: src/main/java/com/iexec/core/feign/SmsClientWrapper.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Service;
+
+import feign.FeignException;
+import lombok.extern.slf4j.Slf4j;
+
+
+@Slf4j
+@Service
+public class SmsClientWrapper {
```
Comment (Ugo) [inline src/main/java/com/iexec/core/feign/SmsClientWrapper.java]: Isn't SmsService better than *Wrapper?

File: src/main/java/com/iexec/core/feign/ResultRepoClientWrapper.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Service;
+
+import feign.FeignException;
+import lombok.extern.slf4j.Slf4j;
+
+
+@Slf4j
+@Service
+public class ResultRepoClientWrapper {
```
Comment (Ugo) [inline src/main/java/com/iexec/core/feign/ResultRepoClientWrapper.java]: Should we call it ResultRepoService instead and move it to the correct package?

File: src/main/java/com/iexec/core/sms/SmsService.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package com.iexec.core.sms;
+
+import com.iexec.common.utils.BytesUtils;
+import com.iexec.core.feign.SmsClientWrapper;
+
+import org.springframework.stereotype.Service;
+
+@Service
+public class SmsService {
```
Comment (Ugo) [inline src/main/java/com/iexec/core/sms/SmsService.java]: Probably SmsService and SmsClientWrapper can be merged.

File: src/main/java/com/iexec/core/feign/SmsClientWrapper.java (hunk 1 above)
Comment (Zied-Guesmi) [reply src/main/java/com/iexec/core/feign/SmsClientWrapper.java]: Actually I thought we'd have a wrapper for each feign client so we handle failure, but if that generates a confusion I'll merge it with the service.

File: src/main/java/com/iexec/core/feign/ResultRepoClientWrapper.java (hunk 2 above)
Comment (Zied-Guesmi) [reply src/main/java/com/iexec/core/feign/ResultRepoClientWrapper.java]: Clearer yes, done it !

File: src/main/java/com/iexec/core/sms/SmsService.java (hunk 3 above)
Comment (Zied-Guesmi) [reply src/main/java/com/iexec/core/sms/SmsService.java]: 👍

**Pull request merged**